rotkehlchen/db/settings.py

```diff
@@ -306,3 +317,96 @@ def serialize_db_setting(
+    submit_usage_analytics: bool = DEFAULT_SUBMIT_USAGE_ANALYTICS
+    active_modules: Sequence[ModuleName] = field(default=DEFAULT_ACTIVE_MODULES)  # type: ignore
+    frontend_settings: str = ''
+    account_for_assets_movements: bool = DEFAULT_ACCOUNT_FOR_ASSETS_MOVEMENTS
+    btc_derivation_gap_limit: int = DEFAULT_BTC_DERIVATION_GAP_LIMIT
+    calculate_past_cost_basis: bool = DEFAULT_CALCULATE_PAST_COST_BASIS
+    display_date_in_localtime: bool = DEFAULT_DISPLAY_DATE_IN_LOCALTIME
+    current_price_oracles: Sequence[CurrentPriceOracle] = field(default=DEFAULT_CURRENT_PRICE_ORACLES)  # noqa: E501
+    historical_price_oracles: Sequence[HistoricalPriceOracle] = field(default=DEFAULT_HISTORICAL_PRICE_ORACLES)  # noqa: E501
+    taxable_ledger_actions: Sequence[LedgerActionType] = field(default=DEFAULT_TAXABLE_LEDGER_ACTIONS)  # noqa: E501
+    pnl_csv_with_formulas: bool = DEFAULT_PNL_CSV_WITH_FORMULAS
+    pnl_csv_have_summary: bool = DEFAULT_PNL_CSV_HAVE_SUMMARY
+    ssf_graph_multiplier: int = DEFAULT_SSF_GRAPH_MULTIPLIER
+    last_data_migration: int = DEFAULT_LAST_DATA_MIGRATION
+    non_syncing_exchanges: Sequence[ExchangeLocationID] = field(default_factory=list)
+    cost_basis_method: CostBasisMethod = DEFAULT_COST_BASIS_METHOD
+    treat_eth2_as_eth: bool = DEFAULT_TREAT_ETH2_AS_ETH
+    eth_staking_taxable_after_withdrawal_enabled: bool = DEFAULT_ETH_STAKING_TAXABLE_AFTER_WITHDRAWAL_ENABLED  # noqa: 501
+    address_name_priority: Sequence[AddressNameSource] = DEFAULT_ADDRESS_NAME_PRIORITY
+    include_fees_in_cost_basis: bool = DEFAULT_INCLUDE_FEES_IN_COST_BASIS
+    infer_zero_timed_balances: bool = DEFAULT_INFER_ZERO_TIMED_BALANCES
+    query_retry_limit: int = DEFAULT_QUERY_RETRY_LIMIT
+    connect_timeout: int = DEFAULT_CONNECT_TIMEOUT
+    read_timeout: int = DEFAULT_READ_TIMEOUT
+
+
+class CachedSettings:
+    """
+    Singleton class that manages the cached settings. It is initialized with default values on
+    user login and is reset on user logout. This way the cached settings are bound to the user's
+    session and not shared between users. It is updated when a setting is updated.
+
+    Keep in mind that settings that are not attributes of the DBSettings dataclass and are
+    not set with the set_setting method are not being cached. Settings not in the DBSettings
+    but set with the set_setting method are cached.
+    """
+    __instance: Optional['CachedSettings'] = None
+    _settings: DBSettingsUnfrozen = DBSettingsUnfrozen()  # the default settings values
+
+    def __new__(cls) -> 'CachedSettings':
+        if CachedSettings.__instance is not None:
+            return CachedSettings.__instance
+
+        CachedSettings.__instance = super().__new__(cls)
+        return CachedSettings.__instance
+
+    def reset(self) -> None:
+        self._settings = DBSettingsUnfrozen()
+
+    def update_entry(self, attr: str, value: Any) -> None:
+        if value is None:
```

Comment:
This is very bad python logic. You are throwing the type checker out of the window with the `Any` type and instead do a dynamic if check every time this is called. Use proper types here and lose the dynamic checks.